### PR TITLE
chore(lambda-nodejs): unescaped whitespace in props.bundling.inject breaks esbuild

### LINKED ISSUE #29559: aws-lambda-nodejs: incorrectly maps esbuild arguments when there is whitespace in path for inject
### Describe the bug

When building NodejsFunctions using esbuild, it incorrectly maps the arguments to esbuild if there are paths in `props.bundling.inject` that has whitespace.

```
new NodejsFunction(scope, `some-id`, {
   bundling: {
      inject: ['/path with spaces/test.js']
   }
})
```

https://github.com/aws/aws-cdk/blob/0fee99b473c17210a156d74263227cdf3274d253/packages/aws-cdk-lib/aws-lambda-nodejs/lib/bundling.ts#L245

### Expected Behavior

should be wrapped in quotes so it is not treated as multiple inputs.
`-inject:"/path with spaces/test.js"`

### Current Behavior

there are unescaped whitespaces treating it as multiple inputs
`-inject:/path with spaces/test.js`

### Reproduction Steps

attempt to build NodejsFunction with paths with spaces
```
new NodejsFunction(scope, `some-id`, {
   bundling: {
      inject: ['/path with spaces/test.js']
   }
})
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

n/a

### Framework Version

_No response_

### Node.js Version

v18.19.1

### OS

mac os

### Language

TypeScript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by bombguy**
### Issue # (if applicable)
Closes #29559 .

### Reason for this change
NodejsFunction fails to build in projects with spaces in path

### Description of changes
Updated how it maps esbuild arguments from bundling props

### Description of how you validated changes
Tried building the NodejsFunction

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by msambol** [comment]
@bombguy possible to add an integration test for this?

**Review by aaythapa — COMMENTED** [review]
Thanks for making this change! +1 to adding integration tests then we can merge it in

**Comment by aaythapa** [comment]
> > @bombguy possible to add an integration test for this?
> 
> can you point me to where the integration tests are for this package?

I'd assume [here](https://github.com/aws/aws-cdk/tree/main/packages/%40aws-cdk-testing/framework-integ/test/aws-lambda-nodejs), you can find out how to make and run integ tests using [this doc](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)

File: packages/@aws-cdk-testing/framework-integ/test/aws-lambda-nodejs/test/integ.function.ts [hunk 1]
```diff
@@ -37,6 +37,13 @@ class TestStack extends Stack {
       runtime: STANDARD_NODEJS_RUNTIME,
     });
 
+    new lambda.NodejsFunction(this, 'js-handler-bundling-path', {
+      entry: path.join(__dirname, 'integ-handlers/js-handler.js'),
+      bundling: {
+        inject: [path.join(__dirname, 'whitespace path/shim.js')],
+      },
+    });
```
Comment (msambol) [inline packages/@aws-cdk-testing/framework-integ/test/aws-lambda-nodejs/test/integ.function.ts:45]: @bombguy Did this not result in a snapshot change?

**Comment by aaythapa** [comment]
hi @bombguy, looks like some snapshots changed which is causing the build failure. You can use `yarn integ --update-on-failed` to re-make the snapshots

**Comment by bombguy** [comment]
`yarn integ --update-on-failed` was failing due to bad build artifacts. I was able to update the snapshots after a clean build

**Pull request merged**